### ISSUE TITLE
RX 470,480,570,580 no longer supports OpenCL on ROCm 4.5

### ISSUE THREAD

**Issue by boxerab** [RadeonOpenCompute/ROCm#1659]
cf: https://github.com/RadeonOpenCompute/ROCm/issues/1608

And **please don't close this issue** until we have a clear answer - has polaris support been intentionally dropped from ROCm
after only 6 years, or is this an error ?

**Comment by ROCmSupport** [comment]
AFAIK, we have not removed any code intentionally. But maybe something changed in the stack and we don't validate gfx8 on ROCm, so it might not be working anymore.
One thing from support point of view, each card has some duration of support. We can not continue supporting cards for more number of years as per business standards. As new cards coming into the  market, we keep adding the new ones into the supported list and keep dropping the old ones after certain amount of time, which is the process.

Anyhow, I am not closing this ticket right now. Let me wait for some time.
Thank you.

**Comment by boxerab** [comment]
@ROCmSupport  the RX 590 was released in 2018. This is way too early to drop support.

https://www.techpowerup.com/gpu-specs/radeon-rx-590.c3322

**Comment by xuhuisheng** [comment]
Maybe you mean OpenCL. The MIOpen run succeed only with a small patch.

**Comment by crazylinuxer** [comment]
@boxerab I solved this issue by using a docker image of certain version for tensorflow-rocm, when I had tried to use a tensorflow with rx570. Here's my repo with a few little scripts that I used to automate this process: https://github.com/crazylinuxer/tensorflow-rocm-legacy
Maybe this will be helpful for you

**Comment by boxerab** [comment]
@kvirikroma I need OpenCL support. That's gone now.

**Comment by Umio-Yasuno** [comment]
Is it possible that this commit is the cause?
https://github.com/ROCm-Developer-Tools/ROCclr/commit/16044d1b30b822bb135a389c968b8365630da452

**Comment by boxerab** [comment]
> Is it possible that this commit is the cause? [ROCm-Developer-Tools/ROCclr@16044d1](https://github.com/ROCm-Developer-Tools/ROCclr/commit/16044d1b30b822bb135a389c968b8365630da452)

This looks exactly like where OpenCL was disabled. Is it difficult to
build ROCclr with this patch reverted ?

**Comment by xuhuisheng** [comment]
I can have a try do patch ROCclr for OpenCL, but I am not familiar with OpenCL, could you show me a demo for testing?

**Comment by boxerab** [comment]
> I can have a try do patch ROCclr for OpenCL, but I am not familiar with OpenCL, could you show me a demo for testing?

those examples look good, also there is a `clinfo` binary in the `/opt/rocm/opencl/bin` folder, I believe, that can quickly tell
 you if the card is recognized as an opencl device.

**Comment by Atemu** [comment]
And intentionally so.

**Comment by Atemu** [comment]
Has anyone tried to set `ROC_ENABLE_PRE_VEGA` to true? Default seems to be false.

**Comment by boxerab** [comment]
hmmm, that's an interesting idea

**Comment by boxerab** [comment]
@ROCmSupport can you comment on Polaris support in recently released ROCm 5.0 ?

**Comment by MathiasMagnus** [comment]
Dear @ROCmSupport,

> We can not continue supporting cards for more number of years as per business standards.

Nvidia recently dropped Kepler support in CUDA in June 2021, when Kepler was released 2012 April. That's 9 years of support. RX 470 was released June 2016, so ~5 years. A bit over half as much. People who bought an RX 590 (released in 2018 November, easily on shelves throughout 2019) only got 2 years of support. **2 years of support!**

I'm teaching GPGPU to physicists at university and we have a BYOD policy (teaching OpenCL, HIP/CUDA, SYCL) and easily the students who suffer the most are those sporting AMD hardware (myself included, running an RX 580 laptop). It's increasingly hard to install and run any of these APIs. Everything stems from the spotty gfx803 support moved to "partial support" (whatever that means) way too early. AMD shouldn't release products they won't support.

_(FWIW even in my professional capacity it's becoming harder to justify recommending CDNA products, due to all of them being gfx9XYZ variants for such a long time. I can't say with a straight face that MI100s/MI200s will not share the same fate as RX 590s, that MI200 successors won't sport a new ISA and gfx9XYZ will be dropped ever so swiftly, being an ancient ISA flavor. Professional Fiji owners have been burned like this before, not just consumer card owners.)_

My experience trying to get gfx803 working:

- HIP doesn't work with DKMS or with Ubuntu 20.04 upstream kernel
```
mate@GL702ZC:~$ /opt/rocm/bin/rocm-smi


======================= ROCm System Management Interface =======================
================================= Concise Info =================================
GPU  Temp   AvgPwr   SCLK    MCLK     Fan  Perf  PwrCap  VRAM%  GPU%  
0    55.0c  14.253W  608Mhz  2000Mhz  0%   auto  68.0W     7%   0%    
================================================================================
WARNING:  		 One or more commands failed
============================= End of ROCm SMI Log ==============================
mate@GL702ZC:~$ sudo /opt/rocm/bin/rocminfo 
ROCk module is loaded
hsa api call failure at: /long_pathname_so_that_rpms_can_package_the_debug_info/src/rocminfo/rocminfo.cc:1143
Call returned HSA_STATUS_ERROR_OUT_OF_RESOURCES: The runtime failed to allocate the necessary resources. This error may also occur when the core runtime library needs to spawn threads or create internal OS-specific events.
```
- OpenCL ROCr runtime doesn't detect the device, even with `ROC_ENABLE_PRE_VEGA` set.
- OpenCL orca runtime mostly works but
  - crashes inside the runtime when creating the interop context.
  - Failed to finalize BRIG when having `work_group_reduce_add` function in use inside the kernel. (Missing device built-in)

**Comment by dagrim** [comment]
I have OpenCL working in Ubuntu 20.04 with an RX570. Here is what I did :
- install ROCm 4.5.2
- get the 21.10 Radeon Software for Linux for Ubuntu 20.04 .tar.xz from here : https://www.amd.com/en/support/kb/release-notes/rn-amdgpu-unified-linux-21-10
- extract the .tar.xz
- extract the `opencl-orca-amdgpu-pro-icd_21.10-1247438_amd64.deb` and `ocl-icd-libopencl1-amdgpu-pro_21.10-1247438_amd64.deb` files
- check that you get those 4 files : `opt/amdgpu-pro/lib/x86_64-linux-gnu/{libamdocl12cl64.so,libamdocl-orca64.so,libOpenCL.so.1.2,libOpenCL.so.1}`
- put those 4 files into a custom directory, say `/opt/myopencl/lib/`
- create a new file `/etc/OpenCL/vendors/myopencl.icd` with only 1 line : `/opt/myopencl/lib/libamdocl-orca64.so`
- try `clinfo` ; here is what I get : 
```
Number of platforms                               2
  Platform Name                                   AMD Accelerated Parallel Processing
  Platform Vendor                                 Advanced Micro Devices, Inc.
  Platform Version                                OpenCL 2.1 AMD-APP (3224.4)
  Platform Profile                                FULL_PROFILE
  Platform Extensions                             cl_khr_icd cl_amd_event_callback cl_amd_offline_devices 
  Platform Host timer resolution                  1ns
  Platform Extensions function suffix             AMD

  Platform Name                                   Clover
  Platform Vendor                                 Mesa
  Platform Version                                OpenCL 1.1 Mesa 21.2.0-devel (git-eb6d990 2021-05-04 focal-oibaf-ppa)
  Platform Profile                                FULL_PROFILE
  Platform Extensions                             cl_khr_icd
  Platform Extensions function suffix             MESA

  Platform Name                                   AMD Accelerated Parallel Processing
Number of devices                                 1
  Device Name                                     Ellesmere
  Device Vendor                                   Advanced Micro Devices, Inc.
  Device Vendor ID                                0x1002
  Device Version                                  OpenCL 1.2 AMD-APP (3224.4)
  Driver Version                                  3224.4
  Device OpenCL C Version                         OpenCL C 1.2 
  Device Type                                     GPU
  Device Board Name (AMD)                         AMD Radeon RX 570 Series
```

**Comment by dagrim** [comment]
You're right on both points : 
1. ROCm is probably not needed, indeed. Any OpenCL icd would work.
2. The Platform version advertises OpenCL 2.1, and the device version 1.2. In fact, I did not even notice since that solution allowed my OpenCL workflow to run successfully.

**Comment by xuhuisheng** [comment]
I think I used wrong package, after recompile rocm-opencl-runtime, clinfo can display gfx803 device.
I will upload my package in this week, everyone who is interest in opencl and gfx803 can have a test.

**Comment by boxerab** [comment]
> I think I used wrong package, after recompile rocm-opencl-runtime, clinfo can display gfx803 device. I will upload my package in this week, everyone who is interest in opencl and gfx803 can have a test.
> 
> BTW, My card is RX580, test on ubuntu-20.04 and ROCm-5.0.0.

Thanks. If you can document what you did to recompile the runtime, we can try it.

**Comment by Umio-Yasuno** [comment]
https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime#building

> 
> Follow these steps:
> 
> -   Build ROCclr first. Follow the steps in the following link to build ROCclr
>    [ROCclr Readme](https://github.com/ROCm-Developer-Tools/ROCclr)
>    In this step, $OPENCL_DIR and $ROCclr_DIR are defined.
> 
> -   Building OpenCL
> Run these commands:
> 
> ```bash
> cd "$OPENCL_DIR"
> mkdir -p build; cd build
> cmake -DUSE_COMGR_LIBRARY=ON -DCMAKE_PREFIX_PATH="$ROCclr_DIR/build;/opt/rocm/" ..
> make -j$(nproc)
> ```
> 
> Note: For release build, add "-DCMAKE_BUILD_TYPE=Release" to the cmake command line.

**Comment by boxerab** [comment]
> https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime#building
> 
> > Follow these steps:
> > 
> > * Build ROCclr first. Follow the steps in the following link to build ROCclr
> >   [ROCclr Readme](https://github.com/ROCm-Developer-Tools/ROCclr)
> >   In this step, $OPENCL_DIR and $ROCclr_DIR are defined.
> > * Building OpenCL
> >   Run these commands:
> > 
> > ```shell
> > cd "$OPENCL_DIR"
> > mkdir -p build; cd build
> > cmake -DUSE_COMGR_LIBRARY=ON -DCMAKE_PREFIX_PATH="$ROCclr_DIR/build;/opt/rocm/" ..
> > make -j$(nproc)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Note: For release build, add "-DCMAKE_BUILD_TYPE=Release" to the cmake command line.

Did you change any settings or variables before building ?

**Comment by Umio-Yasuno** [comment]
> > https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime#building
> > > Follow these steps:
> > > 
> > > * Build ROCclr first. Follow the steps in the following link to build ROCclr
> > >   [ROCclr Readme](https://github.com/ROCm-Developer-Tools/ROCclr)
> > >   In this step, $OPENCL_DIR and $ROCclr_DIR are defined.
> > > * Building OpenCL
> > >   Run these commands:
> > > 
> > > ```shell
> > > cd "$OPENCL_DIR"
> > > mkdir -p build; cd build
> > > cmake -DUSE_COMGR_LIBRARY=ON -DCMAKE_PREFIX_PATH="$ROCclr_DIR/build;/opt/rocm/" ..
> > > make -j$(nproc)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Note: For release build, add "-DCMAKE_BUILD_TYPE=Release" to the cmake command line.
> 
> Did you change any settings or variables before building ?

Sorry, I haven't tried it in my environment yet.  
At least, I think ROCclr and ROCm-OpenCL-Runtime need to rebuilt to enable OpenCL for gfx803, .

**Comment by FelixSchwarz** [comment]
@xuhuisheng Thank you for your comment. It encouraged me to try again and it works now. I'm not sure why this is as I just recompiled my RPMs but `clinfo` now detects my gfx803 device as well. :-)

**Comment by xuhuisheng** [comment]
@FelixSchwarz 
I rechecked building scripts, when we compile rocm-opencl-runtime, is will trigger compiling ROCclr, So It shouldn't effect the result.

My problem is, at first time, I am wrong to copmile HIP, not ROCm-OpenCL-Runtime.

**Comment by FelixSchwarz** [comment]
sorry for spamming here but I just found out why patching `ROC_ENABLE_PRE_VEGA` did not have an effect for me previously: I needed to install `rocm-comgr` as well. I happened to build all of my RPMs with `mock` so I missed comgr even though it was present at build time.

**Comment by FelixSchwarz** [comment]
> My problem is, at first time, I am wrong to compile HIP, not ROCm-OpenCL-Runtime.

@xuhuisheng Maybe you can check the work [Debian's ROCm team](https://salsa.debian.org/rocm-team/) is doing and/or use their [mailing list](https://lists.debian.org/debian-ai/). Fedora is present there as well.

**Comment by Atemu** [comment]
@SolitaryDragon that's completely unrelated to this issue. Please put it in an appropriate place instead (a new issue for example) and don't ping all of us who are waiting on updates on this specific issue.

**Comment by sampie** [comment]
There should be a list of supported cards and the period during which the card is being supported so that users would know the expected end date of the support for their card. And I also agree that is way too early for Polaris to go out of support as it is widely being used.

**Comment by John-Gee** [comment]
If the patch works, can it be made into a PR and accepted by AMD? @ROCmSupport

**Comment by lamdalamda** [comment]
That's why there is still few developers outside AMD trying to migrate their softwares to ROCm after it has been released for five years. I have a gfx803 test platform for coding and was planning to migrate the CUDA code to hip so that I can use the juicy FP64 performance of Mi instinct. I spend several hours changing the linux kernel, uninstall the old ROCm and dealing with all stupid conflict. I was that ambitious until I notice that the ROCm 5.0 no longer support gfx803 and I can never perform local test to debug and see if my migration works. Who the hell will ever do the migration work on the HPC platform? Who paid for the GPU hours? Now I feel I am literally a clown. Come on, you are trying to persuade people to switch to a new platform which is 10 years younger, your MI250 is a game changer but your FP64 is 0 FLOP if I cannot run my software on it. Do something.

**Comment by BloodyMess** [comment]
> I had uploaded patched rocm-opencl-runtime package to github. https://github.com/xuhuisheng/rocm-gfx803/releases/download/rocm500/rocm-opencl_2.0.0-local_amd64.deb
> 
> The patch file: https://github.com/xuhuisheng/rocm-build/blob/feature/build/patch/31.rocm-opencl-runtime-rocclr-gfx803-1.patch Just change ROC_ENABLE_PRE_VEGA from false to true.
> 
> The build script: https://github.com/xuhuisheng/rocm-build/blob/feature/build/31.rocm-opencl-runtime.sh
> 
> And please make sure you are testing on ubuntu-20.04 with ROCm-5.0.0 and gfx803 card.
> 
> `/opt/rocm/bin/opencl/bin/clinfo`
> detail

Has anyone gotten this building and working on an ubuntu 18.04 based system?

My reason is that I'm on elementaryOS 5.1.7 (18.04 based) and am wary of trying to upgrade to the latest 6.x release because the only "upgrade" path they support from 5.1.7 is to do a clean install.

My video card is a RX590.

**Comment by CosmicFusion** [comment]
> sorry for spamming here but I just found out why patching `ROC_ENABLE_PRE_VEGA` did not have an effect for me previously: I needed to install `rocm-comgr` as well. I happened to build all of my RPMs with `mock` so I missed comgr even though it was present at build time.

So from what I understand that doesn't work because because AMD forgot/temporarily disabled/disabled this Flag for some reason so if AMD was to enabled it Polaris would "Un-Officially Supported" rather the being "Dropped" is that correct or Am I not understanding it right ? Plz answer

**Comment by CosmicFusion** [comment]
> @xuhuisheng Thank you for your comment. It encouraged me to try again and it works now. I'm not sure why this is as I just recompiled my RPMs but `clinfo` now detects my gfx803 device as well. :-)

Could you make a guide on how to build this for fedora or at least send me an RPM ? Plz

**Comment by FelixSchwarz** [comment]
@CosmicFusion very little free time now but basically you can use https://copr.fedorainfracloud.org/coprs/fschwarz/rocm-f35/builds/ + a patch for rocm-opencl from https://copr.fedorainfracloud.org/coprs/mystro256/rocm-opencl/ . I guess it's best if you send me an email so I don't have to spam this bug tracker.

**Comment by BloodyMess** [comment]
I've got DaVinci Resolve 17.4.5 build 7 mostly working on elementary 5.1.7 (ubuntu 18.04), with a radeon RX590.

Blender Cycles is working fine now.

Here's a demo of the results:
https://youtu.be/u_TQaArfROE

**Comment by bluescarni** [comment]
> I had uploaded patched rocm-opencl-runtime package to github. https://github.com/xuhuisheng/rocm-gfx803/releases/download/rocm500/rocm-opencl_2.0.0-local_amd64.deb
> 
> The patch file: https://github.com/xuhuisheng/rocm-build/blob/feature/build/patch/31.rocm-opencl-runtime-rocclr-gfx803-1.patch Just change ROC_ENABLE_PRE_VEGA from false to true.
> 
> The build script: https://github.com/xuhuisheng/rocm-build/blob/feature/build/31.rocm-opencl-runtime.sh
> 
> And please make sure you are testing on ubuntu-20.04 with ROCm-5.0.0 and gfx803 card.
> 
> `/opt/rocm/bin/opencl/bin/clinfo`
> detail

Thanks a lot for the patch! It fixed the detection of my RX 570 on my Gentoo system.

**Comment by keryell** [comment]
> AFAIK, we have not removed any code intentionally. But maybe something changed in the stack and we don't validate gfx8 on ROCm, so it might not be working anymore. One thing from support point of view, each card has some duration of support. We can not continue supporting cards for more number of years as per business standards. As new cards coming into the market, we keep adding the new ones into the supported list and keep dropping the old ones after certain amount of time, which is the process.

This is open-source software and thus this should follow the spirit of mind openness.
There is a difference between supporting something officially and forbidding something explicitly instead of just allowing it as a best effort and welcoming outside contributions to make things just work™.
Something working, even if not supported, is better than something not working. :-)

**Comment by John-Gee** [comment]
Thanks to xuhuisheng 's patch, Arch's rocm now enables again Pre Vega cards (only on AUR so far, but there is hope arch4edu will pick it up and then eventually community).

**Comment by stefanharjes** [comment]
> > I had uploaded patched rocm-opencl-runtime package to github. https://github.com/xuhuisheng/rocm-gfx803/releases/download/rocm500/rocm-opencl_2.0.0-local_amd64.deb
> > The patch file: https://github.com/xuhuisheng/rocm-build/blob/feature/build/patch/31.rocm-opencl-runtime-rocclr-gfx803-1.patch Just change ROC_ENABLE_PRE_VEGA from false to true.
> > The build script: https://github.com/xuhuisheng/rocm-build/blob/feature/build/31.rocm-opencl-runtime.sh
> > And please make sure you are testing on ubuntu-20.04 with ROCm-5.0.0 and gfx803 card.
> > `/opt/rocm/bin/opencl/bin/clinfo`
> > detail
> 
> Thanks a lot for the patch! It fixed the detection of my RX 570 on my Gentoo system.

could you please elaborate? did you manually install or emerge the rocm-opencl-runtime-5.0.2? When I patch the ebuild,
I do not get clinto to recognize my gfx803.

Thansk

**Comment by CosmicFusion** [comment]
hi everyone , i think i have a definitive answer , although AMD isn't communicating with us properly , they seem to have told the fedora team , that they want to support gfx803 , but are too busy on other things , so they dropped it from official to expermintal 
see : https://fedoraproject.org/wiki/SIGs/HC#HW_Support
so if you want to use your gfx803 gpu you need to : 

```
sudo echo   'ROC_ENABLE_PRE_VEGA=1' >> /etc/environment
```
and no patch is needed.

i have tested this on my RX 580 and it work in opencl & hip.

i also spoke with the blender team. 

since they choose to not put gfx803 in their hip targets

they said that there is a possibility for them to make it work ,
but they don't wanna because : 
"WE DON'T WANT TO PUT THE IMPRESSION THAT IT IS ON THE ROADMAP SINCE IT'S NOT OFFICIALLY SUPPORTED BY AMD."

**Comment by Mhowser** [comment]
> so if you want to use your gfx803 gpu you need to :
> 
> `sudo echo   'ROC_ENABLE_PRE_VEGA=1' >> /etc/environment`
> and no patch is needed.
Is this exclusive to Fedora only? Or can we use this on other distros?

**Comment by CosmicFusion** [comment]
> > so if you want to use your gfx803 gpu you need to :
> > 
> > `sudo echo   'ROC_ENABLE_PRE_VEGA=1' >> /etc/environment`
> > and no patch is needed.
> 
> 
> Is this exclusive to Fedora only? Or can we use this on other distros?
> 
> 
> 

Can be used on all distros , tested on
 PopOS using repo.radeon.com , and fedora using their packages , and my personal built from source ones

**Comment by BloodyMess** [comment]
> > > so if you want to use your gfx803 gpu you need to :
> > > `sudo echo   'ROC_ENABLE_PRE_VEGA=1' >> /etc/environment`
> > > and no patch is needed.
> > 
> > 
> > Is this exclusive to Fedora only? Or can we use this on other distros?
> 
> Can be used on all distros , tested on PopOS using repo.radeon.com , and fedora using their packages , and my personal built from source ones

The page you linked on Fedora says (I added bold-italic to highlight):

> Most products starting with vega 10 (Radeon RX Vega) should work without configuration, but any pre-vega HW, such as the RX 400's or 500's, will require setting (_**note this is no longer required for rocm-opencl-5.2.1-2 or later**_): 

Can anyone confirm this is correct for the latest ROCm releases?

**Comment by xuhuisheng** [comment]
I test on ROCm-5.2.3 + RX580, and it works.

I think it is an undocumented environment variable to controll whether we can use OPENCL on gfx803.

**Comment by BloodyMess** [comment]
> I test on ROCm-5.2.3 + RX580, and it works.
> 
> I think it is an undocumented environment variable to controll whether we can use OPENCL on gfx803.

Yes, I agree with that.

What I think is happening regarding that phrase on [the Fedora page](https://fedoraproject.org/wiki/SIGs/HC#HW_Support), is that Fedora is doing their ROCm builds from 5.2.1-2 onward with that variable defaulting to true (or maybe their build package sets the environment variable when the installation is done?).

So, the rest of us non-Fedora users have to continue setting that environment variable.

**Comment by Mhowser** [comment]
Is anybody able to test if HIP could be used with this method as well?

**Comment by CosmicFusion** [comment]
> Is anybody able to test if HIP could be used with this method as well?

HIP doesn't directly depend on ROCclr , and thus works on gfx803 ootb , and I can run hip code with it ,  but forcing blender to render on it caused a white render , so there is an issue on the blender side that needs fixing

**Comment by Mhowser** [comment]
That's exactly the use case I wanted to test myself, I couldn't even get Blender to detect my RX 580 yet.

**Comment by xuhuisheng** [comment]
@CosmicFusion 
hi, recently, I just find out the blender is a 3d designer app, as 3d max.
And because of my rx580 character , I do some dig for blender on gfx803 with ROCm. Unfortunately, even latest blender-3.3 said it can support gfx900, gfx803 always export a abnormal image.
Compiling OK, export image wont get right result.

**Comment by CosmicFusion** [comment]
> @CosmicFusion 
> hi, recently, I just find out the blender is a 3d designer app, as 3d max.
> And because of my rx580 character , I do some dig for blender on gfx803 with ROCm. Unfortunately, even latest blender-3.3 said it can support gfx900, gfx803 always export a abnormal image.
> Compiling OK, export image wont get right result.

Me and GloriousEggroll tried adding gfx803 to blender 3.4 , it either renders a white image or a red mess , this is blender issue not a hip one as I can compile hip code successfully on my RX 580

**Comment by rajhlinux** [comment]
[CosmicFusion](https://github.com/CosmicFusion)

> > > so if you want to use your gfx803 gpu you need to :
> > > `sudo echo   'ROC_ENABLE_PRE_VEGA=1' >> /etc/environment`
> > > and no patch is needed.
> > 
> > 
> > Is this exclusive to Fedora only? Or can we use this on other distros?
> 
> Can be used on all distros , tested on PopOS using repo.radeon.com , and fedora using their packages , and my personal built from source ones

You think I can use OpenCL 1.2 on FreeBSD 13.1 for my RX-580 GPU?

**Comment by redthing1** [comment]
Wow, I got it to work using https://github.com/rocm-arch/rocm-arch/blob/master/README.md and `HSA_OVERRIDE_GFX_VERSION=10.3.0`

**Comment by owariee** [comment]
> AFAIK, we have not removed any code intentionally. But maybe something changed in the stack and we don't validate gfx8 on ROCm, so it might not be working anymore. One thing from support point of view, each card has some duration of support. We can not continue supporting cards for more number of years as per business standards. As new cards coming into the market, we keep adding the new ones into the supported list and keep dropping the old ones after certain amount of time, which is the process.
> 
> Anyhow, I am not closing this ticket right now. Let me wait for some time. Thank you.

i dont seen this being a valid response from AMD as a consumer, i never, really, never, seen a driver for a series of cards dropping support for a feature that is marketed in the cards on purpose, so really, fix this ASAP.  Community are handling your issues for you, on arch polaris works just fine with a patched package from AUR (https://archlinux.org/packages/community-testing/x86_64/rocm-opencl-runtime/)

**Comment by John-Gee** [comment]
> > AFAIK, we have not removed any code intentionally. But maybe something changed in the stack and we don't validate gfx8 on ROCm, so it might not be working anymore. One thing from support point of view, each card has some duration of support. We can not continue supporting cards for more number of years as per business standards. As new cards coming into the market, we keep adding the new ones into the supported list and keep dropping the old ones after certain amount of time, which is the process.
> > Anyhow, I am not closing this ticket right now. Let me wait for some time. Thank you.
> 
> i dont seen this being a valid response from AMD as a consumer, i never, really, never, seen a driver for a series of cards dropping support for a feature that is marketed in the cards on purpose, so really, fix this ASAP. Community are handling your issues for you, on arch polaris works just fine with a patched package from AUR ([archlinux.org/packages/community-testing/x86_64/rocm-opencl-runtime](https://archlinux.org/packages/community-testing/x86_64/rocm-opencl-runtime/))

You don't need to patch ROCm anymore, ROC_ENABLE_PRE_VEGA=1 should be enough.

**Comment by redthing1** [comment]
@CosmicFusion I attempted blender port as well. I have successfully run the 3.3 ROCm build. But when I used the same patches for 3.4 I got the same thing you said, some white and red mess.

**Comment by CosmicFusion** [comment]
> @CosmicFusion I attempted blender port as well. I have successfully run the 3.3 ROCm build. But when I used the same patches for 3.4 I got the same thing you said, some white and red mess.

wait do you mean you were able to successfully render on blender using 3.3

**Comment by Mhowser** [comment]
> > @CosmicFusion I attempted blender port as well. I have successfully run the 3.3 ROCm build. But when I used the same patches for 3.4 I got the same thing you said, some white and red mess.
> 
> wait do you mean you were able to successfully render on blender using 3.3?

@redthing1 what is the hardware that you used?

**Comment by CosmicFusion** [comment]
> > > @CosmicFusion I attempted blender port as well. I have successfully run the 3.3 ROCm build. But when I used the same patches for 3.4 I got the same thing you said, some white and red mess.
> > 
> > wait do you mean you were able to successfully render on blender using 3.3?
> 
> @redthing1 what is the hardware that you used?

Gigabyte RX 580 8GB with only the ROC_ENABLE_PRE_VEGA=1, for the record this still works in ROCm 5.4.1, but i haven't tested HIP since 5.2.3 and blender 3.4 alpha or beta i don't remember, and that was without the HSA OVERRIDE env, I currently have exams so i can't test

**Comment by redthing1** [comment]
> > > @CosmicFusion I attempted blender port as well. I have successfully run the 3.3 ROCm build. But when I used the same patches for 3.4 I got the same thing you said, some white and red mess.
> > 
> > 
> > wait do you mean you were able to successfully render on blender using 3.3?
> 
> @redthing1 what is the hardware that you used?



> > > > @CosmicFusion I attempted blender port as well. I have successfully run the 3.3 ROCm build. But when I used the same patches for 3.4 I got the same thing you said, some white and red mess.
> > > 
> > > 
> > > wait do you mean you were able to successfully render on blender using 3.3?
> > 
> > 
> > @redthing1 what is the hardware that you used?
> 
> Gigabyte RX 580 8GB with only the ROC_ENABLE_PRE_VEGA=1, for the record this still works in ROCm 5.4.1 for OpenCL, but i haven't tested HIP since 5.2.3 and blender 3.4 alpha or beta i don't remember, and that was without the HSA OVERRIDE env, I currently have exams so i can't test
> 
> EDIT : Ohh you are not talking to me lol

Well funny enough. He used the exact same hardware as me. Gigabyte RX 580 8GB with only the ROC_ENABLE_PRE_VEGA=1

**Comment by boxerab** [comment]
> Well funny enough. He used the exact same hardware as me. Gigabyte RX 580 8GB with only the ROC_ENABLE_PRE_VEGA=1

@redthing1 what Linux distribution are you using ? And how do you set ROC_ENABLE_PRE_VEGA=1 ?  Do you have to set this variable before you install ?

**Comment by redthing1** [comment]
> > Well funny enough. He used the exact same hardware as me. Gigabyte RX 580 8GB with only the ROC_ENABLE_PRE_VEGA=1
> 
> @redthing1 what Linux distribution are you using ? And how do you set ROC_ENABLE_PRE_VEGA=1 ? Do you have to set this variable before you install ?

Yes

**Comment by boxerab** [comment]
> > > Well funny enough. He used the exact same hardware as me. Gigabyte RX 580 8GB with only the ROC_ENABLE_PRE_VEGA=1
> > 
> > 
> > @redthing1 what Linux distribution are you using ? And how do you set ROC_ENABLE_PRE_VEGA=1 ? Do you have to set this variable before you install ?
> 
> Yes

Great! I tried running install script from 
https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.4.3/page/How_to_Install_ROCm.html
but I didn't have the env variable set and it failed to find the cl device. I guess I could set it first and then try again.

**Comment by boxerab** [comment]
Alright, I admit defeat. Tried to install 5.4.3 on Ubuntu 22 from install script with `ROC_ENABLE_PRE_VEGA` env variable set in `/etc/environment`. No cl device detected.

Fedora 37 works right out of the box, so I will stick with Fedora.

Unfortunately, the performance of my cl kernels is 50% of what I used to get with ROCm 3.x

**Comment by CosmicFusion** [comment]
> Alright, I admit defeat. Tried to install 5.4.3 on Ubuntu 22 from install script with `ROC_ENABLE_PRE_VEGA` env variable set in `/etc/environment`. No cl device detected.
> 
> Fedora 37 works right out of the box, so I will stick with Fedora.
> 
> Unfortunately, the performance of my cl kernels is 50% of what I used to get with ROCm 3.x

You need to run this
```
sudo usermod -aG video $LOGNAME
sudo usermod -aG render $LOGNAME
reboot

```

**Comment by boxerab** [comment]
> You need to run this
> 
> ```
> sudo usermod -aG video $LOGNAME
> sudo usermod -aG render $LOGNAME
> reboot
> ```

Thanks,  I will sick with Fedora. I am assuming that perf won't change between the two distributions.

**Comment by boxerab** [comment]
As a matter of fact, running my kernels on Fedora 37 with latest packaged ROCm gives same performance as with 3.x ROCm.

I am quite happy with the Fedora 37 packages and their support for Polaris cards. 
Going to close this.

**Comment by Mhowser** [comment]
So we must use Fedora if we want to use our cards for anything compute related? Can these packages be ported over to other distros?

**Comment by boxerab** [comment]
> So we must use Fedora if we want to use our cards for anything compute related? Can these packages be ported over to other distros?

Some here have got it working for officially supported Ubuntu 22 with AMD packages, unfortunately I only had luck with Fedora. Also I believe there is support on Arch. For other distros I'm not sure.

**Comment by tsl0922** [comment]
> > So we must use Fedora if we want to use our cards for anything compute related? Can these packages be ported over to other distros?
> 
> Some here have got it working for officially supported Ubuntu 22 with AMD packages, unfortunately I only had luck with Fedora. Also I believe there is support on Arch. For other distros I'm not sure.

I can confirm that rocm-5.4.3 works on Ubuntu 22.04.2 LTS / Python 3.10.6 for stable-diffusion-webui, using a manually compiled pytorch 1.13.1 with `PYTORCH_ROCM_ARCH=gfx803`.

https://github.com/tsl0922/pytorch-gfx803

**Comment by vittorio88** [comment]
@tsl0922 Hey, so it seem that with `ROCM_ENABLE_PRE_VEGA` enabled, RX580 and family works just fine?

Why is it that TORCH requires to be manually compiled? 
Just to add: `export PYTORCH_ROCM_ARCH=gfx803` ?

Why is this not the default for Torch, and do you know if/where this was changed in Torch code?
